src/components/HomeMap/HomeMap.tsx

```diff
@@ -91,6 +96,100 @@ export const HomeMap = ({ routes }: { routes: GeoJSON.FeatureCollection }) => {
         },
       });
 
+      // Add start and end markers for routes
+      const { startMarkers, endMarkers } = createRouteMarkersData(routes);
+
+      // Add start markers (triangles)
+      map.addSource("start-markers", {
+        type: "geojson",
+        data: startMarkers,
+      });
+
+      // Add end markers (perpendicular lines)
+      map.addSource("end-markers", {
+        type: "geojson",
+        data: endMarkers,
+      });
+
+      // Load all marker images asynchronously
+      const loadMarkerImages = async () => {
+        const imagePromises: Promise<void>[] = [];
+
+        // Generate and add start marker images
+        startMarkers.features.forEach((feature, index) => {
+          const bearing = feature.properties?.bearing || 0;
+          const iconId = `start-triangle-${index}`;
+          const dataUrl = generateTriangleSVG(bearing, 16, "#059669"); // green color for start
+
+          const promise = new Promise<void>((resolve) => {
```

Comment:
you can use `loadImageToMap` helper to load the image, but avoid doing it in the loop. Load images once and they loop over the routes